src/dmd/expressionsem.d

```diff
@@ -10225,6 +10225,24 @@ private extern (C++) final class ExpressionSemanticVisitor : Visitor
             return setError();
         }
 
+        if (t1.ty == Tclass && t2.ty == Tclass)
+        {
+            // Lower to object.__cmp(e1, e2)
+            Expression cl = new IdentifierExp(exp.loc, Id.empty);
```

Comment:
Will this lower to `object.__cmp(e1, e2)` or `.object.__cmp(e1, e2)`? Note the leading dot.

Reply:
Is there a `.object.__cmp(e1, e2)` ? I assumed that `object.__cmp(e1, e2)` gives the full path, so the dot(`.`) for "take the global `__cmp(e1, e2)`" wouldn't change who gets called. Am I wrong?

Reply:
`object.__cmp(e1, e2)` could find a local class or instance `object` and search it for a member `__cmp`, but `Id.empty` takes care of starting the search at module level. So `.object.__cmp(e1, e2)` would be a more accurate comment, but that is not done anywhere else.

Reply:
> but that is not done anywhere else

It doesn't hurt to improve. I had to ask so it was not clear enough.